spacetemplate/repository.go

```diff
@@ -0,0 +1,286 @@
+package spacetemplate
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/gormsupport"
+	"github.com/almighty/almighty-core/log"
+	"github.com/almighty/almighty-core/space"
+	"github.com/almighty/almighty-core/workitem"
+	"github.com/almighty/almighty-core/workitem/link"
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"golang.org/x/net/context"
+)
+
+// Repository describes interactions with space templates
+type Repository interface {
+	Create(ctx context.Context, st *SpaceTemplate) error
+	List(ctx context.Context) ([]SpaceTemplate, error)
+	Load(ctx context.Context, spaceTemplateID uuid.UUID) (*SpaceTemplate, error)
+	// Apply tries to load the space template id as stored in the space and then
+	// creates all the artifacts like work item type and work item link types
+	Apply(ctx context.Context, spaceTemplateID uuid.UUID, spaceID uuid.UUID) error
```

Comment:
In principle you are right. But I would have to load the space in order to get the space template from it. And that is something I cannot do in all cases. Consider the case in which I verify a space template by creating a space (with `deleted_at != nil`). I wouldn't be able to load the temporary space. That's why I've decided to pass in both, the space template ID and the space ID.

Also, in the end it's the space template that I need to load and not the space, whereas if I would just have the space ID I would need to load the space and then the space template. So all in all, I'm saving one load-